src/components/v5/common/AvatarWithAddress/AvatarWithAddress.tsx

```diff
@@ -0,0 +1,64 @@
+import { CircleWavyCheck, WarningCircle } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import { isAddress } from 'ethers/lib/utils';
+import React, { type ReactNode, type FC } from 'react';
+
+import { formatText } from '~utils/intl.ts';
+import UserAvatar, {
+  type UserAvatarProps,
+} from '~v5/shared/UserAvatar/index.ts';
+
+interface AvatarWithAddressProps
+  extends Omit<UserAvatarProps, 'userAddress' | 'size'> {
+  address: string;
+  isVerified?: boolean;
+  title?: ReactNode | string;
+  size?: UserAvatarProps['size'];
+}
+const AvatarWithAddress: FC<AvatarWithAddressProps> = ({
+  address,
+  isVerified,
+  userName,
+  userAvatarSrc,
+  size = 20,
+  title,
+  ...rest
+}) => {
+  const isUserAddressValid = isAddress(address);
+  return isUserAddressValid ? (
+    <>
+      <UserAvatar
```

Comment:
do you think we could just compose the avatar in, so this component doesn't need to care about the size and whatnot? so we would just do `{avatar}` even if we need to pass the `address` to both `UserAvatar` and `AvatarWithAddress`?
Then we could also potentially reuse this for the colony and token avatars too!

Reply:
Thank you for your comments, I just decided that this AvatarWithAddress looks more complex than I thought, so I will create a separate ticket to refactor it. And I reverted this changes for this PR